### DISCUSSION TITLE
How to do backups with document exporter?

### DISCUSSION THREAD

**Discussion by niarbx** [jonaswinkler/paperless-ng#282] [Support] [answered]
Hi all,

I hope somebody can point me in the right direction. I'm trying to figure out how the document exporter is supposed to be used. I'm trying to use the exporter to make daily backups of my documents. As Jonas statet in antother post (https://github.com/jonaswinkler/paperless-ng/discussions/135), he is using `rsync --link-dest` to keep the used disk space down.
The problem I have, is that the document exporter seems to not be able to overwrite (or even better, append to) existing exports.
So if my export directory is not empty, document exporter will abort telling me so.
So I wrote a little script which basically does the following:
1. clear the export folder /paperless/export
2. export all the documents to this folder (with document_exporter)
3. create new directory within my backup directory, eg /backup/2020-01-06
4. rsync the exported files to this new directory with the following command

`rsync -avPShyH --link-dest=/backup/2021-01-05 /paperless/export/ /backup/2021-01-06`

The exported files in both of the directories (2021-01-05 and 2021-01-06) are the same, since there was no change in documents in these two days.
Sadly, rsync doesnt recognize this fact and does no linking but creates a whole new backup. I suppose because the files where newly created and now have different attributes. 
Or do I have some other errors in my thinking, the rsync command or even the whole backup process?

Thanks a lot for your help.

Best regards

**Comment by jonaswinkler** [comment, marked as answer]
There's no error in your thinking. 

> I suppose because the files where newly created and now have different attributes.

Exactly. I'm using rsync on the docker volumes directly, not doing anything with the document exporter, that's why I didn't recognize this as an issue.

I'll modify the exporter to support this scenario. It should update the export directory with new contents, leaving unmodified files untouched, just as you said.

**Comment by niarbx** [comment]
Hi Jonas,

Ok, this makes sense then. I'm doing backups of the data and media directories as well. I just thought, better be safe then sorry and do additional backups of the PDF files with real filenames :) 

Then I'm good for the moment. Thanks for the quick answer!

**Reply by jonaswinkler** [reply]
You could tell paperless to use real filenames in the media directory as well!

https://paperless-ng.readthedocs.io/en/latest/advanced_usage.html#file-name-handling

**Reply by niarbx** [reply]
Oh great, thanks for the hint. I will have a look at it, maybe this solves the whole export thing.
